public/app/core/logsModel.ts

```diff
@@ -733,7 +722,24 @@ export function queryLogsVolume<TQuery extends DataQuery, TOptions extends DataS
           });
           observer.error(error);
         } else {
-          rawLogsVolume = rawLogsVolume.concat(dataQueryResponse.data.map(toDataFrame));
+          const aggregatedLogsVolume = aggregateRawLogsVolume(
+            dataQueryResponse.data.map(toDataFrame),
+            options.extractLevel
+          );
+          if (aggregatedLogsVolume[0]) {
+            aggregatedLogsVolume[0].meta = {
+              custom: {
+                targets: options.targets,
+                absoluteRange: { from: options.range.from.valueOf(), to: options.range.to.valueOf() },
+              },
+            };
+          }
+          rawLogsVolume = aggregatedLogsVolume;
+          observer.next({
+            state: LoadingState.Streaming,
```

Comment:
Did you check if using `LoadingState.Loading` had any visible impact in the UI?

Reply:
Doesn't change anything, but in b82d606 I added a small indicator:

![image](https://user-images.githubusercontent.com/8092184/218080892-d2606b51-da58-469b-8d1b-46a79424e48d.png)

WDYT?

Reply:
Wunderbar!